Cabal/tests/UnitTests/Distribution/Version.hs

```diff
@@ -218,7 +217,9 @@ prop_withinVersion v v' =
      withinRange v' (withinVersion v)
   == (v' >= v && v' < upper v)
   where
-    upper (Version lower t) = Version (init lower ++ [last lower + 1]) t
+    upper vlower = mkVersion (init lower ++ [last lower + 1])
+      where
+        lower = unVersion vlower
```

Comment:
Alright, in the latest patch I've exposed `alterVersion` so we can use it in the test-suite